core/domain/user_services.py

```diff
@@ -1390,6 +1523,19 @@ def set_username(user_id: str, new_username: str) -> None:
     save_user_settings(user_settings)
 
 
+def get_all_users_usernames() -> List[str]:
```

Comment:
We shouldn't be doing this. See above.

Reply:
Updated

Reply:
Er, we still should not be doing this. There should be no function here that does a get_all() on tens to hundreds of thousands of users and iterates over them. That takes too long.

Reply:
Oops! Sorry, I thought that I removed it. Updated now.